src/interface/viewport.c

```diff
@@ -2243,7 +2257,10 @@ rct_xy16 screen_coord_to_viewport_coord(rct_viewport *viewport, uint16 x, uint16
 rct_xy16 viewport_coord_to_map_coord(int x, int y, int z)
 {
 	rct_xy16 ret;
-	switch (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32)) {
+	uint32 rotation = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32);
+	switch (rotation) {
+	default:
+		log_warning("Improper rotation %u", rotation);
```

Comment:
Something like calling this could be used everywhere.
```c
int get_current_rotation()
{
    int rotation = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_ROTATION, uint32);
#if DEBUG
    if (rotation > 3) {
        // AHHH, log!
    }
#endif
    return rotation & 3;
}
```